examples/java/src/main/java/org/apache/beam/examples/WordCount.java

```diff
@@ -180,8 +181,11 @@ static void runWordCount(WordCountOptions options) {
     p.apply("ReadLines", TextIO.read().from(options.getInputFile()))
         .apply(new CountWords())
         .apply(MapElements.via(new FormatAsTextFn()))
-        .apply("WriteCounts", TextIO.write().to(options.getOutput()));
-
+        .apply(
+            "WriteCounts",
+            options.as(StreamingOptions.class).isStreaming()
```

Comment:
The `--streaming` flag really just controls whether a runner is in "streaming mode". Not all runners even have a streaming mode, and it should never affect the semantics of the pipeline. So I don't think that should affect the code or logic of the example. The problem seems to be a bug in the `DataflowRunner`, not anything to do with the example.

Reply:
looks like the StreamingShardedWriteFactory overwrites the WriteFiles transform(which incorporates Create.Values() that should be overwritten by StreamingFnApiCreateOverrideFactory, I've tested reordering of the overrides and it seems to solve the problem.

it seems hard in general to keep the ordering always correct if we only replace all transforms in one run but I guess the Factories that overwrites into a composite transform should be added first.

Reply:
Yea it is hard to keep the ordering correct. I have some troubles with it all. We probably need to have a bigger plan for ordering them. Or, we could just apply them over and over until it stops... (or times out)